### PR TITLE
Issue #54: Enhancement - Flatten arrays

### PR DESCRIPTION
**Pull request by codetheorist**
Fixes issue #54 

* Add `getRawData()` function to `SettingStore.php`
* Update README.md to reflect this new feature

settings-screenshot

### CODE REVIEW

File: src/SettingStore.php [hunk 1]
```diff
@@ -48,6 +48,29 @@ public function get($key, $default = null)
...
 
 	/**
+	 * Get a flattened array from the settings data.
+	 *
+	 * @param  array   $array   Optional array of settings values, with a fallback to all settings.
+	 * @param  string  $prefix  Optional prefix.
+	 *
+	 * @return array
+	 */
+	public function getRawData($array = null, $prefix = null)
```
Comment (anlutro) [inline src/SettingStore.php]: I think the function should be named "getFlattenedData" instead, as "raw" doesn't really convey what it does.

File: src/SettingStore.php [hunk 2]
```diff
@@ -48,6 +48,29 @@ public function get($key, $default = null)
 	}
 
 	/**
+	 * Get a flattened array from the settings data.
+	 *
+	 * @param  array   $array   Optional array of settings values, with a fallback to all settings.
```
Comment (anlutro) [inline src/SettingStore.php]: This is a bit confusing. Maybe make an optional `$keys` argument instead? Similar to how `Request::only` works.

File: src/SettingStore.php [hunk 3]
```diff
@@ -48,6 +48,29 @@ public function get($key, $default = null)
...
+	 * @param  string  $prefix  Optional prefix.
```
Comment (anlutro) [inline src/SettingStore.php]: I would expand the description a bit. "Optional prefix for keys in the returned array" or something like that.

File: src/SettingStore.php (hunk 1 above)
Comment (anlutro) [reply src/SettingStore.php]: Actually, just `flattened` would probably be best, to be consistent with other method names (`all` and `get`).

File: src/SettingStore.php [hunk 4]
```diff
@@ -48,6 +48,29 @@ public function get($key, $default = null)
...
+	 *
+	 * @return array
+	 */
+	public function getRawData($array = null, $prefix = null)
+	{
+		$subject = isset($array) ? $array : \Setting::all();
```
Comment (bweston92) [inline src/SettingStore.php]: You're assuming everyone has the facade setup here. `s/\Setting/Facade/`

File: src/SettingStore.php [hunk 5]
```diff
@@ -48,6 +48,29 @@ public function get($key, $default = null)
...
+		$result = array();
+		foreach($subject as $key=>$value) {
+			if(is_array($value)) {
+				$result = $result + flatten($value, $prefix . $key . '.');
+			}
+			else {
```
Comment (bweston92) [inline src/SettingStore.php]: Use PSR style if/else blocks - https://github.com/anlutro/laravel-settings/commit/7e67a94f99a00f4d6161608e02b0e223190acbb2#diff-6a3371457528722a734f3c51d9238c13R5

**Comment by bweston92** [comment]
@codetheorist any update?

File: src/SettingStore.php (hunk 5 above)
Comment (gpressutto5) [reply src/SettingStore.php]: Why using else at all?

**Comment by bweston92** [comment]
Closing for inactivity, feel free to open again.

**Comment by thewebartisan7** [comment]
What do you think of this:

```php


    /**
     * Get a flattened array from the settings data.
     *
     * @param  array|null $data Optional array of settings values, with a fallback to all settings.
     * @param  string $prefix
     *
     * @return array
     */
    public function flattened($data = null, $prefix = null)
    {
        if(is_null($data)) {
            $this->load();
            $data = $this->data;
        }

        $result = array();
        foreach($data as $key => $value) {
            if(is_array($value)) {
                $result = $result + $this->flattened($value, "{$prefix}{$key}.");
            } else {
                $result["{$prefix}{$key}"] = $value;
            }
        }
        return $result;
    }

```


I can make a pull request and try make some tests.

**Pull request closed without merging**